### PR TITLE
feat(app): add bare bones protocols list to ODD

### PR DESCRIPTION
**Pull request by shlokamin**
# Overview

This PR adds a bare bones protocols list page to the ODD app. It [does not look like it does in Figma](https://www.figma.com/file/AoTLAYuWawlaWItB1umOjr/OT-3-On-Device-Display?node-id=1780%3A27844&t=WiDiZMKmCkAg0YLb-1), but has all the right data.

I'll do the same sort of thing with the protocol details page, and we can spruce up the presentation later.

# Changelog

- Add barebones protocols list to ODD

# Review requests

Go to the protocols list page on the ODD, you should see a list of protocols with a:
 - protocol name
 - last run
 - date added

Note, in order to have protocols populate the protocols list, they need to... exist. that means you need to first either
 - use the `send to OT-3` button in the overflow menu from the desktop app if you're testing on an ODD
 - run a protocol on your dev server if you're testing on your computer (this is because starting a run also creates a protocol record)

# Risk assessment
Low

### CODE REVIEW

File: app/src/pages/OnDeviceDisplay/ProtocolDashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+          {protocols.data?.data.map((protocol, index) => {
+            const lastRun = runs.data?.data.find(
+              run => run.protocolId === protocol.id
+            )?.createdAt
+            return (
+              <TableRow
+                key={index}
+                onClick={() => history.push(`/protocols/${protocol.id}`)}
+              >
+                <TableDatum> {protocol.metadata.protocolName}</TableDatum>
```
Comment (brenthagen) [inline app/src/pages/OnDeviceDisplay/ProtocolDashboard.tsx]: can we do a filename fallback here? 
`protocol.metadata.protocolName ?? protocol.files[0].name`

**Review by brenthagen — APPROVED** [review]
✔️

File: app/src/pages/OnDeviceDisplay/ProtocolDashboard.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            const lastRun = runs.data?.data.find(
+              run => run.protocolId === protocol.id
+            )?.createdAt
+
+            const protocolName =
+              protocol.metadata.protocolName ?? protocol.files[0].name
+
+            return (
+              <TableRow
+                key={index}
```
Comment (koji) [inline app/src/pages/OnDeviceDisplay/ProtocolDashboard.tsx]: For desktop, protocols landing page uses `protocolKey` for key, so maybe using `protocol.id` would be good. But not necessary

**Review by koji — APPROVED** [review]
Recommend to add a test to OnDeviceDisplayApp.test.tsx` since it won't be affected when we update protocols list.

File: app/src/pages/OnDeviceDisplay/ProtocolDashboard.tsx (hunk 2 above)
Comment (shlokamin) [reply app/src/pages/OnDeviceDisplay/ProtocolDashboard.tsx]: i think we have to use the protocol id to be able to fetch the protocol details from the server (on the OT-3)

on the desktop app, i think we need the protocol key to look for local analysis but im not actually sure, maybe we can also use id

Comment (koji) [reply app/src/pages/OnDeviceDisplay/ProtocolDashboard.tsx]: sorry the description wasn't clear. I mean `key` prop.

```suggestion
  key={protocol.id}
```

**Pull request merged**